src/BaseFirestoreTransactionRepository.ts

```diff
@@ -56,7 +56,7 @@ export class TransactionRepository<T extends IEntity>
       item.id = doc.id;
     }
 
-    await this.transaction.set(doc, this.toSerializableObject(item as T));
+    this.transaction.set(doc, this.toSerializableObject(item as T));
```

Comment:
By removing the await in this function we're not waiting for the operation to finish. That might bring errors like trying to access a model that was not yet created in firestore or getting an error randomly in another function because the promise was rejected. We must await for the operation to end before getting out this function.

Reply:
According to TypeScript, `set` doesn't return a promise :)

Reply:
As far as I can tell, `set` doesn't return a promise? See [here](https://github.com/googleapis/nodejs-firestore/blob/master/types/firestore.d.ts#L311)

Reply:
Oh, looking at the source code, the transaction methods run synchronously because intertally it's managed by a writebatch.